Modules/Sources/PointOfSale/Presentation/Orders/POSOrderDetailsView.swift

```diff
@@ -377,59 +376,98 @@ private extension POSOrderDetailsView {
 
 // MARK: - Actions
 private extension POSOrderDetailsView {
-    enum POSOrderDetailsAction: Identifiable, CaseIterable {
+    enum POSAction: Identifiable, CaseIterable {
+        case issueRefund
         case emailReceipt
 
         var id: String { title }
 
         var title: String {
             switch self {
-            case .emailReceipt:
-                Localization.emailReceiptActionTitle
+            case .issueRefund:  Localization.issueRefundActionTitle
+            case .emailReceipt: Localization.emailReceiptActionTitle
+            }
+        }
+
+        var accessibilityHint: String {
+            switch self {
+            case .issueRefund:  Localization.issueRefundAccessibilityHint
+            case .emailReceipt: Localization.emailReceiptAccessibilityHint
+            }
+        }
+
+        var priority: Int {
+            switch self {
+            case .issueRefund:  100
+            case .emailReceipt: 50
             }
         }
 
-        func available(for order: POSOrder) -> Bool {
+        func isAvailable(for order: POSOrder, flags: POSFeatureFlagProviding) -> Bool {
+            guard order.status == .completed else { return false }
             switch self {
+            case .issueRefund:
+                return flags.isFeatureFlagEnabled(.pointOfSaleRefundsi1)
             case .emailReceipt:
-                order.status == .completed
+                return true
+            }
+        }
+    }
+
+    func handler(for action: POSAction) -> @MainActor () -> Void {
+        switch action {
+        case .emailReceipt:
+            return {
+                analytics.track(event: WooAnalyticsEvent.PointOfSale.orderDetailsEmailReceiptTapped())
+                isShowingEmailReceiptView = true
             }
+        case .issueRefund:
+            return { }
         }
     }
 
-    var actions: [POSOrderDetailsAction] {
-        POSOrderDetailsAction.allCases.filter { $0.available(for: order) }
+    var availableActions: [POSAction] {
+        POSAction.allCases
+            .filter { $0.isAvailable(for: order, flags: featureFlags) }
+            .sorted { $0.priority > $1.priority }
     }
 
     @ViewBuilder
-    func actionsSection(_ actions: [POSOrderDetailsAction]) -> some View {
-        VStack {
-            HStack {
-                ForEach(actions) { action in
-                    Button(action: {
-                        switch action {
-                        case .emailReceipt:
-                            analytics.track(event: WooAnalyticsEvent.PointOfSale.orderDetailsEmailReceiptTapped())
-                            isShowingEmailReceiptView = true
+    func actionsSection(actions: [POSAction]) -> some View {
+        if actions.isEmpty {
+            EmptyView()
+        } else {
+            HStack(spacing: POSSpacing.large) {
+                let primary = actions[0]
```

Comment:
Let's be explicit on which action is triggering rather than rely on array order, using the specific enum case or others.

Reply:
As we talked, the logic is dynamic, so we need some data structure to define the actions at runtime. That said, I added a struct to hold that here 1083d9ad0827703503ab0c792d901ecd23b2326c, which should improve the readability compared to an array.